src/jobs/actions/urlaction.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  Logger,
+  NotFoundException,
+  BadRequestException,
+  HttpException,
+} from "@nestjs/common";
+import { JobAction } from "../config/jobconfig";
+import { JobClass } from "../schemas/job.schema";
+import * as Handlebars from "handlebars";
+
+// Handlebar options for JobClass templates
+// TODO should this be moved into job.schema.ts?
+const jobTemplateOptions = {
+  allowedProtoProperties: {
+    id: true,
+    type: true,
+    statusCode: true,
+    statusMessage: true,
+    messageSent: true,
+    createdBy: true,
+    jobParams: false,
+  },
+  allowProtoPropertiesByDefault: false, // limit accessible fields for security
```

Comment:
I went with a more restrictive setting for now while trying to understand the security. See #1113 for details.

This approach is harder to maintain since the templatable properties need to be kept up to date with the JobClass schema.